kv/kvserver/gc: TestEndToEndGC failed

kv/kvserver/gc.TestEndToEndGC [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7837350?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7837350?buildTab=artifacts#/) on master @ [8a5cb5109d6b7e122e996a44f940bfd61698cd81](https://github.com/cockroachdb/cockroach/commits/8a5cb5109d6b7e122e996a44f940bfd61698cd81):


```
=== RUN   TestEndToEndGC
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/f27dc6f41a7e04b7d5e45680c447408b/logTestEndToEndGC3981558965
    test_log_scope.go:79: use -show-logs to present logs inline
    gc_int_test.go:186: range 54 stats: contains_estimates:0 last_update_nanos:1670234911600483106 intent_age:0 gc_bytes_age:0 live_bytes:43994 live_count:1000 key_bytes:24994 key_count:1000 val_bytes:19000 val_count:1000 intent_bytes:0 intent_count:0 separated_intent_count:0 range_key_count:0 range_key_bytes:0 range_val_count:0 range_val_bytes:0 sys_bytes:384 sys_count:6 abort_span_bytes:0 
    gc_int_test.go:196: 
        	Error Trace:	/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2177/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/gc/gc_test_/gc_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvserver/gc/gc_int_test.go:196
        	Error:      	Should be empty, but was [313604547210643 6674366349807930 15469134800960042 51298362238944579 60163324744386291]
        	Test:       	TestEndToEndGC
        	Messages:   	table still contains data after range deletion
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/f27dc6f41a7e04b7d5e45680c447408b/logTestEndToEndGC3981558965
--- FAIL: TestEndToEndGC (8.30s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestEndToEndGC.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-22114

kv/kvserver/gc.TestEndToEndGC [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7925525?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7925525?buildTab=artifacts#/) on master @ [942a4d468e9c8ad0ef45a7be33f0a326dfb19fef](https://github.com/cockroachdb/cockroach/commits/942a4d468e9c8ad0ef45a7be33f0a326dfb19fef):


```
=== RUN   TestEndToEndGC
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/f27dc6f41a7e04b7d5e45680c447408b/logTestEndToEndGC1293893962
    test_log_scope.go:79: use -show-logs to present logs inline
    gc_int_test.go:186: range 54 stats: contains_estimates:0 last_update_nanos:1670836000909136315 intent_age:0 gc_bytes_age:0 live_bytes:36990 live_count:1000 key_bytes:24990 key_count:1000 val_bytes:12000 val_count:1000 intent_bytes:0 intent_count:0 separated_intent_count:0 range_key_count:0 range_key_bytes:0 range_val_count:0 range_val_bytes:0 sys_bytes:509 sys_count:7 abort_span_bytes:0 
    gc_int_test.go:196: 
        	Error Trace:	/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2611/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/gc/gc_test_/gc_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvserver/gc/gc_int_test.go:196
        	Error:      	Should be empty, but was [335517240515191 5228955155085247 19940095233228208 26828138299905875 28727452027240922]
        	Test:       	TestEndToEndGC
        	Messages:   	table still contains data after range deletion
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/f27dc6f41a7e04b7d5e45680c447408b/logTestEndToEndGC1293893962
--- FAIL: TestEndToEndGC (8.32s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestEndToEndGC.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

kv/kvserver/gc.TestEndToEndGC [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7987110?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/7987110?buildTab=artifacts#/) on master @ [dcdf599e9eaee8010137167f419509ccb7627406](https://github.com/cockroachdb/cockroach/commits/dcdf599e9eaee8010137167f419509ccb7627406):


```
=== RUN   TestEndToEndGC
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/f27dc6f41a7e04b7d5e45680c447408b/logTestEndToEndGC2170054627
    test_log_scope.go:79: use -show-logs to present logs inline
    gc_int_test.go:186: range 54 stats: contains_estimates:0 last_update_nanos:1671180082395219092 intent_age:0 gc_bytes_age:0 live_bytes:36998 live_count:1000 key_bytes:24998 key_count:1000 val_bytes:12000 val_count:1000 intent_bytes:0 intent_count:0 separated_intent_count:0 range_key_count:0 range_key_bytes:0 range_val_count:0 range_val_bytes:0 sys_bytes:370 sys_count:6 abort_span_bytes:0 
    gc_int_test.go:196: 
        	Error Trace:	/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2699/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/gc/gc_test_/gc_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvserver/gc/gc_int_test.go:196
        	Error:      	Should be empty, but was [7183179638478131 62992632699256750 95750077917060772 104850500708940552 108137850389009922]
        	Test:       	TestEndToEndGC
        	Messages:   	table still contains data after range deletion
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/f27dc6f41a7e04b7d5e45680c447408b/logTestEndToEndGC2170054627
--- FAIL: TestEndToEndGC (7.66s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestEndToEndGC.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

kv/kvserver/gc.TestEndToEndGC [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8002268?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8002268?buildTab=artifacts#/) on master @ [7c8e7012b73a598f3aa1da9bbf422e833706f5d3](https://github.com/cockroachdb/cockroach/commits/7c8e7012b73a598f3aa1da9bbf422e833706f5d3):


```
=== RUN   TestEndToEndGC
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/f27dc6f41a7e04b7d5e45680c447408b/logTestEndToEndGC4263871640
    test_log_scope.go:79: use -show-logs to present logs inline
    gc_int_test.go:186: range 54 stats: contains_estimates:0 last_update_nanos:1671267645254383402 intent_age:0 gc_bytes_age:0 live_bytes:43993 live_count:1000 key_bytes:24993 key_count:1000 val_bytes:19000 val_count:1000 intent_bytes:0 intent_count:0 separated_intent_count:0 range_key_count:0 range_key_bytes:0 range_val_count:0 range_val_bytes:0 sys_bytes:532 sys_count:7 abort_span_bytes:0 
    gc_int_test.go:196: 
        	Error Trace:	/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2585/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/gc/gc_test_/gc_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvserver/gc/gc_int_test.go:196
        	Error:      	Should be empty, but was [3251167650446708 11455444954257485 15026282430323365 41076431492388830 54527374662286785]
        	Test:       	TestEndToEndGC
        	Messages:   	table still contains data after range deletion
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/f27dc6f41a7e04b7d5e45680c447408b/logTestEndToEndGC4263871640
--- FAIL: TestEndToEndGC (7.43s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestEndToEndGC.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

kv/kvserver/gc.TestEndToEndGC [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8008937?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8008937?buildTab=artifacts#/) on master @ [93ed65565357538c9048ff45c878a493f2ed9b45](https://github.com/cockroachdb/cockroach/commits/93ed65565357538c9048ff45c878a493f2ed9b45):


```
=== RUN   TestEndToEndGC
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/f27dc6f41a7e04b7d5e45680c447408b/logTestEndToEndGC1941637116
    test_log_scope.go:79: use -show-logs to present logs inline
    gc_int_test.go:186: range 54 stats: contains_estimates:0 last_update_nanos:1671354227802293599 intent_age:0 gc_bytes_age:0 live_bytes:43987 live_count:1000 key_bytes:24987 key_count:1000 val_bytes:19000 val_count:1000 intent_bytes:0 intent_count:0 separated_intent_count:0 range_key_count:0 range_key_bytes:0 range_val_count:0 range_val_bytes:0 sys_bytes:250 sys_count:6 abort_span_bytes:0 
    gc_int_test.go:196: 
        	Error Trace:	/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2346/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/gc/gc_test_/gc_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvserver/gc/gc_int_test.go:196
        	Error:      	Should be empty, but was [3029303936593393 13365432440079643 23211926943461559 23800527842682014 29859519143774313]
        	Test:       	TestEndToEndGC
        	Messages:   	table still contains data after range deletion
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/f27dc6f41a7e04b7d5e45680c447408b/logTestEndToEndGC1941637116
--- FAIL: TestEndToEndGC (7.43s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestEndToEndGC.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

kv/kvserver/gc.TestEndToEndGC [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8016220?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8016220?buildTab=artifacts#/) on master @ [94e03c016955dfd64d4e15358ea92226a8f362aa](https://github.com/cockroachdb/cockroach/commits/94e03c016955dfd64d4e15358ea92226a8f362aa):


```
=== RUN   TestEndToEndGC
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/f27dc6f41a7e04b7d5e45680c447408b/logTestEndToEndGC131303321
    test_log_scope.go:79: use -show-logs to present logs inline
    gc_int_test.go:186: range 54 stats: contains_estimates:0 last_update_nanos:1671439728555671286 intent_age:0 gc_bytes_age:0 live_bytes:43995 live_count:1000 key_bytes:24995 key_count:1000 val_bytes:19000 val_count:1000 intent_bytes:0 intent_count:0 separated_intent_count:0 range_key_count:0 range_key_bytes:0 range_val_count:0 range_val_bytes:0 sys_bytes:250 sys_count:6 abort_span_bytes:0 
    gc_int_test.go:196: 
        	Error Trace:	/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2563/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/gc/gc_test_/gc_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvserver/gc/gc_int_test.go:196
        	Error:      	Should be empty, but was [16352724286990300 19660061598266018 25721500818412135 61653022085577109 70530124565693345]
        	Test:       	TestEndToEndGC
        	Messages:   	table still contains data after range deletion
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/f27dc6f41a7e04b7d5e45680c447408b/logTestEndToEndGC131303321
--- FAIL: TestEndToEndGC (7.82s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestEndToEndGC.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

kv/kvserver/gc.TestEndToEndGC [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8032931?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8032931?buildTab=artifacts#/) on master @ [61f7c816574848f804f1adf7f42f7315bfcaaeba](https://github.com/cockroachdb/cockroach/commits/61f7c816574848f804f1adf7f42f7315bfcaaeba):


```
=== RUN   TestEndToEndGC
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/f27dc6f41a7e04b7d5e45680c447408b/logTestEndToEndGC782296963
    test_log_scope.go:79: use -show-logs to present logs inline
=== CONT  TestEndToEndGC
    gc_int_test.go:270: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/f27dc6f41a7e04b7d5e45680c447408b/logTestEndToEndGC782296963
--- FAIL: TestEndToEndGC (26.65s)
=== RUN   TestEndToEndGC/rangeTombstones=true/clearRange=false
    gc_int_test.go:216: range 54 stats: contains_estimates:0 last_update_nanos:1671526291885728486 intent_age:0 gc_bytes_age:0 live_bytes:36994 live_count:1000 key_bytes:24994 key_count:1000 val_bytes:12000 val_count:1000 intent_bytes:0 intent_count:0 separated_intent_count:0 range_key_count:0 range_key_bytes:0 range_val_count:0 range_val_bytes:0 sys_bytes:370 sys_count:6 abort_span_bytes:0 
    gc_int_test.go:230: 
        	Error Trace:	/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3105/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/gc/gc_test_/gc_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvserver/gc/gc_int_test.go:230
        	Error:      	Should be empty, but was [780068662832119 8970561606168755 40610098206715408 55364230462907072 64393266184181978]
        	Test:       	TestEndToEndGC/rangeTombstones=true/clearRange=false
        	Messages:   	table still contains data after range deletion
    --- FAIL: TestEndToEndGC/rangeTombstones=true/clearRange=false (7.85s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestEndToEndGC.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

kv/kvserver/gc.TestEndToEndGC [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8047541?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8047541?buildTab=artifacts#/) on master @ [10266a323f94c3cf397d5de590e512d987e63e22](https://github.com/cockroachdb/cockroach/commits/10266a323f94c3cf397d5de590e512d987e63e22):


```
=== RUN   TestEndToEndGC
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/f27dc6f41a7e04b7d5e45680c447408b/logTestEndToEndGC1497645839
    test_log_scope.go:79: use -show-logs to present logs inline
=== CONT  TestEndToEndGC
    gc_int_test.go:270: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/f27dc6f41a7e04b7d5e45680c447408b/logTestEndToEndGC1497645839
--- FAIL: TestEndToEndGC (28.48s)
=== RUN   TestEndToEndGC/rangeTombstones=true/clearRange=false
    gc_int_test.go:216: range 54 stats: contains_estimates:0 last_update_nanos:1671611274425236287 intent_age:0 gc_bytes_age:0 live_bytes:36987 live_count:1000 key_bytes:24987 key_count:1000 val_bytes:12000 val_count:1000 intent_bytes:0 intent_count:0 separated_intent_count:0 range_key_count:0 range_key_bytes:0 range_val_count:0 range_val_bytes:0 sys_bytes:509 sys_count:7 abort_span_bytes:0 
    gc_int_test.go:230: 
        	Error Trace:	/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3118/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/gc/gc_test_/gc_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvserver/gc/gc_int_test.go:230
        	Error:      	Should be empty, but was [422680544915786 1165821809319837 11863241664635494 23933655950705296 24021801462568651]
        	Test:       	TestEndToEndGC/rangeTombstones=true/clearRange=false
        	Messages:   	table still contains data after range deletion
    --- FAIL: TestEndToEndGC/rangeTombstones=true/clearRange=false (7.62s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestEndToEndGC.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

kv/kvserver/gc.TestEndToEndGC [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8063184?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8063184?buildTab=artifacts#/) on master @ [15bc0c4780681d95e705c22dbe2e7d8d190054ef](https://github.com/cockroachdb/cockroach/commits/15bc0c4780681d95e705c22dbe2e7d8d190054ef):


```
=== RUN   TestEndToEndGC
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/f27dc6f41a7e04b7d5e45680c447408b/logTestEndToEndGC3045965912
    test_log_scope.go:79: use -show-logs to present logs inline
=== CONT  TestEndToEndGC
    gc_int_test.go:270: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/f27dc6f41a7e04b7d5e45680c447408b/logTestEndToEndGC3045965912
--- FAIL: TestEndToEndGC (28.94s)
=== RUN   TestEndToEndGC/rangeTombstones=true/clearRange=false
    gc_int_test.go:216: range 54 stats: contains_estimates:0 last_update_nanos:1671697405008081844 intent_age:0 gc_bytes_age:0 live_bytes:36991 live_count:1000 key_bytes:24991 key_count:1000 val_bytes:12000 val_count:1000 intent_bytes:0 intent_count:0 separated_intent_count:0 range_key_count:0 range_key_bytes:0 range_val_count:0 range_val_bytes:0 sys_bytes:243 sys_count:6 abort_span_bytes:0 
    gc_int_test.go:230: 
        	Error Trace:	/home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2430/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/kv/kvserver/gc/gc_test_/gc_test.runfiles/com_github_cockroachdb_cockroach/pkg/kv/kvserver/gc/gc_int_test.go:230
        	Error:      	Should be empty, but was [6842196956073839 17810771739719240 18996097457723007 23888026268739831 25416788034312762]
        	Test:       	TestEndToEndGC/rangeTombstones=true/clearRange=false
        	Messages:   	table still contains data after range deletion
    --- FAIL: TestEndToEndGC/rangeTombstones=true/clearRange=false (7.60s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestEndToEndGC.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/kv/kvserver/gc/gc_int_test.go
+++ b/pkg/kv/kvserver/gc/gc_int_test.go
@@ -144,13 +144,22 @@ func TestEndToEndGC(t *testing.T) {
 					return startKey, endKey
 				}
 
-				deleteRangeDataWithRangeTombstone := func(t *testing.T, rangeIDs ids, kvDb *kv.DB, db *gosql.DB) {
+				deleteRangeDataWithRangeTombstone := func(t *testing.T, kvDb *kv.DB, db *gosql.DB) {
 					t.Helper()
-					for _, id := range rangeIDs {
-						start, end := getRangeInfo(t, id, db)
-						require.NoError(t, kvDb.DelRangeUsingTombstone(ctx, start, end),
-							"failed to delete range with tombstone")
+					var prevRangeIDs ids
+					for i := 0; i < 3; i++ {
+						rangeIDs := getTableRangeIDs(t, sqlDb)
+						if rangeIDs.equal(prevRangeIDs) {
+							return
+						}
+						for _, id := range rangeIDs {
+							start, end := getRangeInfo(t, id, db)
+							require.NoError(t, kvDb.DelRangeUsingTombstone(ctx, start, end),
+								"failed to delete range with tombstone")
+						}
+						prevRangeIDs = rangeIDs
 					}
+					t.Fatal("failed to get consistent list of ranges for table after 3 attempts")
 				}
 
 				getRangeStats := func(t *testing.T, rangeID int64) enginepb.MVCCStats {
@@ -198,6 +207,7 @@ func TestEndToEndGC(t *testing.T) {
 				for i := 0; i < 1000; i++ {
 					execOrFatal(t, sqlDb, "upsert into kv values ($1, $2)", rng.Int63(), "hello")
 				}
+				t.Logf("found table range after initializing table data: %s", getTableRangeIDs(t, sqlDb))
 
 				require.NotEmptyf(t, readSomeKeys(t, sqlDb), "found no keys in table")
 
@@ -219,13 +229,16 @@ func TestEndToEndGC(t *testing.T) {
 					}
 				}
 				require.NotEmpty(t, tableRangeIDs, "failed to query ranges belonging to table")
-				require.NotEmptyf(t, nonEmptyRangeIDs, "all table ranges are empty according to MVCCStats")
+				require.NotEmpty(t, nonEmptyRangeIDs, "all table ranges are empty according to MVCCStats")
+
+				t.Logf("found non-empty table ranges before deletion: %v", nonEmptyRangeIDs)
 
 				if d.rangeTombstones {
-					deleteRangeDataWithRangeTombstone(t, tableRangeIDs, kvDb, sqlDb)
+					deleteRangeDataWithRangeTombstone(t, kvDb, sqlDb)
 				} else {
 					execOrFatal(t, sqlDb, "delete from kv where k is not null")
 				}
+				t.Logf("found table ranges after range deletion: %s", getTableRangeIDs(t, sqlDb))
 
 				require.Empty(t, readSomeKeys(t, sqlDb), "table still contains data after range deletion")
 
@@ -261,6 +274,7 @@ func TestEndToEndGC(t *testing.T) {
 
 					nonEmptyRangeIDs := findNonEmptyRanges(t, tableRangeIDs)
 					if len(nonEmptyRangeIDs) > 0 {
+						t.Logf("non empty ranges after GC queue: %s", nonEmptyRangeIDs)
 						return errors.New("not all ranges were cleared")
 					}
 					return nil
@@ -277,4 +291,16 @@ func (r ids) String() string {
 		s[i] = fmt.Sprintf("%d", r)
 	}
 	return fmt.Sprintf("[%s]", strings.Join(s, ","))
+}
+
+func (r ids) equal(o ids) bool {
+	if len(r) != len(o) {
+		return false
+	}
+	for i := range r {
+		if r[i] != o[i] {
+			return false
+		}
+	}
+	return true
 }

